src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -79,7 +79,7 @@ public function denormalize($data, $class, $format = null, array $context = arra
 
             $dateTimeErrors = \DateTime::class === $class ? \DateTime::getLastErrors() : \DateTimeImmutable::getLastErrors();
 
-            throw new UnexpectedValueException(sprintf(
+            throw new NotNormalizableValueException(sprintf(
```

Comment:
Maybe can we set a specific exception code for the different class of exceptions to be able to easily detect if it's an invalid data: URI, or an invalid date and so on?